Tests/SwiftDocCTests/Infrastructure/BundleDiscoveryTests.swift

```diff
@@ -327,31 +310,24 @@ class BundleDiscoveryTests: XCTestCase {
     }
 
     func testThemeSettingsFound() throws {
-        let workspace = Folder(name: "TestBundle.docc", content:
-            allFiles.map { CopyOfFile(original: $0) } + [
-                TextFile(name: "theme-settings.json", utf8Content: """
-                {
-                  "meta": {},
-                  "theme": {
-                    "colors": {
-                      "text": "#ff0000"
-                    }
-                  },
-                  "features": {}
+        let catalog = Folder(name: "Something.docc", content: [
+            TextFile(name: "theme-settings.json", utf8Content: """
+            {
+              "meta": {},
+              "theme": {
+                "colors": {
+                  "text": "#ff0000"
                 }
-                """),
-            ]
-        )
-
-        let tempURL = try createTemporaryDirectory()
-
-        let workspaceURL = try workspace.write(inside: tempURL)
-        let dataProvider = try LocalFileSystemDataProvider(rootURL: workspaceURL)
-
-        let bundles = try dataProvider.bundles(options: BundleDiscoveryOptions())
+              },
+              "features": {}
+            }
+            """),
+        ])
 
-        XCTAssertEqual(bundles.count, 1)
-        guard let bundle = bundles.first else { return }
+        let fileSystem = try TestFileSystem(folders: [catalog])
```

Comment:
Aside from being easier to read and simpler, are these tests now using the in-memory file system instead of an actual temporary directory? And are they faster?

Reply:
Yes, but it's not going to be all that noticeable on the grand scheme of things.

If I look only at the tests in this file, we can see that:
- `testBundleDiscoveryOptions` and `testBundleFormat` are about 2x-3x faster
- `testFirstBundle` and `testLoadComplexWorkspace` are the same (because they're deprecated tests that haven't changed)
- `testCustomTemplatesFound`, `testNoCustomTemplates`, `testNoInfoPlist`, and `testThemeSettingsFound` are too fast to register. 

 Test name                   | on main | on this branch
-----------------------------|---------|----------------
`testBundleDiscoveryOptions` | 0.015 s | 0.007 s
`testBundleFormat`           | 0.064 s | 0.021 s
`testCustomTemplatesFound`   | 0.018 s | 0.000 s
`testFirstBundle`            | 0.021 s | 0.021 s
`testLoadComplexWorkspace`   | 0.027 s | 0.026 s
`testNoCustomTemplates`      | 0.013 s | 0.000 s
`testNoInfoPlist`            | 0.011 s | 0.000 s
`testThemeSettingsFound`     | 0.016 s | 0.000 s  

Across those 8 tests this saves about 100ms (0.184 s on main compared to 0.088 s on this branch).  

That said, across the entire test suite the difference is still less than a second because very few tests changed to using an in-memory file system and each test is only a few milliseconds speedup.